usersession/agent/rest_api.go

```diff
@@ -349,3 +355,50 @@ func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
 	}
 	return SyncResponse(nil)
 }
+
+func postRebootRequiredNotification(c *Command, r *http.Request) Response {
+	if ok, resp := validateJSONRequest(r); !ok {
+		return resp
+	}
+
+	decoder := json.NewDecoder(r.Body)
+
+	var rebootRequired client.RebootRequiredInfo
+	if err := decoder.Decode(&rebootRequired); err != nil {
+		return BadRequest("cannot decode request body into reboot required notification info: %v", err)
+	}
+
+	// Note that since the connection is shared, we are not closing it.
+	if c.s.bus == nil {
+		return SyncResponse(&resp{
+			Type:   ResponseTypeError,
+			Status: 500,
+			Result: &errorResult{
+				Message: "cannot connect to the session bus",
+			},
+		})
+	}
+
+	summary := fmt.Sprintf(i18n.G("Update available for %q."), rebootRequired.InstanceName)
+	body := i18n.G("Restart the system to update now.")
+	hints := []notification.Hint{
+		notification.WithDesktopEntry("io.snapcraft.SessionAgent"),
+		notification.WithUrgency(notification.LowUrgency),
+	}
+
+	msg := &notification.Message{
+		Title: summary,
+		Body:  body,
+		Hints: hints,
```

Comment:
```suggestion
		Priority: notification.PriorityLow,
```

I think you can drop the hints entirely. The GTK backend will ignore them, and the FDO backend should pick `io.snapcraft.SessionAgent` as a default from the manager, and set urgency from the priority.